staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -700,6 +700,27 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 	if err != nil {
 		return nil, err
 	}
+
+	// Create replicasPathInCustomResource
+	replicasPathInCustomResource := fieldmanager.ResourcePathMappings{}
+	for _, v := range crd.Spec.Versions {
+		subresources, err := apiextensionshelpers.GetSubresourcesForVersion(crd, v.Name)
+		if err != nil {
+			utilruntime.HandleError(err)
+			return nil, fmt.Errorf("the server could not properly serve the CR subresources")
+		}
+		if subresources == nil || subresources.Scale == nil {
```

Comment:
But is the scale handler going to gently ignore these, or is it going to remove their entire entry?

Reply:
We absolutely have to fix that.

Reply:
That's a good point. 

A possible solution would be for the scale handler to always ignore entries with version it doesn't know about, instead of dropping. This would apply for both CRs and other resources.  That would be inconsistent with `Apply`, I think.

Another way would be to somehow limit this behaviour to CRs, by passing some parameter down to the scale handler, so that we have a different logic for custom resources. I am not sure I like that though.

Reply:
I think there is a difference between versions that don't exist and versions that don't have a scale subresource.

Reply:
Right, what about https://github.com/kubernetes/kubernetes/pull/98377/commits/1f1d9387ef0d3051d560b4c2bcbdc6bb7d45db60. 

Basically, I am adding an empty path to all CR versions that don't have `scale`. The assumption is that manage fields entries won't ever have an empty path, so those entries just gets ignored [here](https://github.com/nodo/kubernetes/blob/scale-ownership-tracking-deployment/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/scalehandler.go#L121-L124)